### DISCUSSION TITLE
Type Error Root Layout - Does not satisfy constraint LayoutProps

### DISCUSSION THREAD

**Discussion by eprisr** [vercel/next.js#60191] [Help] [answered]
### Summary

React v18.2.0
Next 14.0.4
TS 5.3.2
@types/react 18.2.42

While trying to deploy, I kept receiving an error message:
`Type error: Type '{ children: ReactNode; }' does not satisfy the constraint 'LayoutProps'.
Property 'types' is missing in type '{ children: ReactNode; }' but required in type 'LayoutProps'.`

The only way I could resolve it was by adding types property to children like so:
`export default function RootLayout({
  children
}: {
  children: ReactNode
  types: ReactNode
}) { ...`

But this isn't what's in the documentation to do. Can anyone help me understand why this happened? I manually added Next.js to an existing project if that detail helps any.

### Additional information

```js
Type error: Type '{ children: ReactNode; }' does not satisfy the constraint 'LayoutProps'.
Property 'types' is missing in type '{ children: ReactNode; }' but required in type 'LayoutProps'.
```

**Comment by icyJoseph** [comment, marked as answer]
Do you have a parallel route called `@types` under `app`?

Perhaps something like this:

<img width="459" alt="Screenshot 2024-01-04 at 09 35 55" src="https://github.com/vercel/next.js/assets/21013447/d89c8bd7-6ab5-4b64-b7c8-acf014860ee9">

That'll trigger this, expected, error:

```console
src/app/layout.tsx
Type error: Type '{ children: ReactNode; }' does not satisfy the constraint 'LayoutProps'.
  Property 'types' is missing in type '{ children: ReactNode; }' but required in type 'LayoutProps'.
```

And that's because of how Parallel Routes work: https://nextjs.org/docs/app/building-your-application/routing/parallel-routes

**Reply by LrAmaral** [reply]
I was having the same issue a few moments ago when I was trying to deploy, and I was trying hard to figure out where is the problem, thanks @icyJoseph for your help.

**Reply by eduvhc** [reply]
Thanks it helped a lot.

**Reply by cybervoid0** [reply]
what?? And where's the solution????? I know that Parallel Routes, so what?

**Reply by leGenti** [reply]
Can some please answer with a solution as well?

**Reply by germanao** [reply]
LoL

**Reply by leGenti** [reply]
For the ones still encountering this error, this is what fixed it for me (In my case I got the error when I wanted to add the locale as a prop).

in layout.tsx I've: 
- Added type Params, which should be a Promise as of next 15 I suppose? 
- Added generateMetadata function

```javascript
import { Geist, Geist_Mono } from "next/font/google";
import "../globals.css";

type Params = Promise<{ locale: string }>;

export async function generateMetadata({params}: {params: Params}) {
  const {locale} = await params;
  console.log(locale);
  return {
    title: "ProjectName",
    description: "ProjectDescription",
  }
}

export default async function RootLayout({children, params}: {children: React.ReactNode, params: Params}) {
  const {locale} = await params;

  return (
    <html lang={locale}>
      <body
        className={`${geistSans.variable} ${geistMono.variable} antialiased`}
      >
        {children}
      </body>
    </html>
  );
}
```

**Reply by fberbert** [reply]
This is the answer, thank you!

> type Params = Promise<{ locale: string }>;
> 
> export async function generateMetadata({params}: {params: Params}) {
>   const {locale} = await params;
>   ...

**Reply by holedev** [reply]
i downgrade version from `15.5.x` to `15.4.x` and it working right ...

**Reply by Anguerran** [reply]
Thank you 😿. You have saved my life! I have a presentation in 24 minutes, but it wouldn't build, and the AI was useless, so I had to do it the old way here.

**Comment by arcria** [comment]
Nide sure you don't accidently have  children: React.ReactDOM, instead of children: React.ReactNode

**Reply by Sedi98** [reply]
downgraded version from 15.5.x to 15.2.8 and its fixed

**Comment by Sedi98** [comment]
downgraded version from 15.5.x to 15.2.8 and its fixed
